plugins/launcher_service_plugin/launcher_service_plugin.cpp

```diff
@@ -0,0 +1,1153 @@
+         "name or ip address of host")
+         ("p2p-listen-address", bpo::value<string>()->default_value("0.0.0.0"),
+         "listen address of incoming p2p connection of nodeos")
+         ("nodeos-cmd", bpo::value<string>()->default_value("./programs/nodeos/nodeos"),
+         "nodeos executable command")
+         ("genesis-file", bpo::value<string>()->default_value("genesis.json"),
+         "path of genesis file")
+         ("abi-serializer-max-time", bpo::value<uint32_t>()->default_value(1000000),
+         "abi serializer max time(in us)")
+         ("log-file-max", bpo::value<uint16_t>()->default_value(8),
+         "max number of stdout/stderr files per node")
+         ("print-http", boost::program_options::bool_switch()->notifier([this](bool e){_my->_config.print_http_request = _my->_config.print_http_response = e;}),
+         "print http request and response")
+         ("default-key", bpo::value<string>()->default_value("5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3"),
+         "default private key for cluster accounts")
+
+         ("base-port", bpo::value<uint16_t>()->default_value(1600),
+         "base port number of clusters")
+         ("cluster-port-span", bpo::value<uint16_t>()->default_value(1000),
+         "length of port range reserved per cluster")
+         ("node-port-span", bpo::value<uint16_t>()->default_value(10),
+         "length of port range reserved per node in a cluster")
+         ("max-nodes-per-cluster", bpo::value<uint16_t>()->default_value(100),
+         "max nodes per cluster")
+         ("max-clusters", bpo::value<uint16_t>()->default_value(30),
+         "max number of clusters")
+         ;
+}
+
+void launcher_service_plugin::plugin_initialize(const variables_map& options) {
+   std::cout << "launcher_service_plugin::plugin_initialize()" << std::endl;
+   try {
+      _my->_config.data_dir = options.at("data-dir").as<std::string>();
+      _my->_config.host_name = options.at("host-name").as<std::string>();
+      _my->_config.p2p_listen_addr = options.at("p2p-listen-address").as<std::string>();
+      _my->_config.nodeos_cmd = options.at("nodeos-cmd").as<std::string>();
+      _my->_config.genesis_file = options.at("genesis-file").as<std::string>();
+      _my->_config.abi_serializer_max_time = fc::microseconds(options.at("abi-serializer-max-time").as<uint32_t>());
+      _my->_config.log_file_rotate_max = options.at("log-file-max").as<uint16_t>();
+      _my->_config.default_key = private_key_type(options.at("default-key").as<std::string>());
+      _my->_config.base_port = options.at("base-port").as<uint16_t>();
+      _my->_config.cluster_span = options.at("cluster-port-span").as<uint16_t>();
+      _my->_config.node_span = options.at("node-port-span").as<uint16_t>();
+      _my->_config.max_nodes_per_cluster = options.at("max-nodes-per-cluster").as<uint16_t>();
+      _my->_config.max_clusters = options.at("max-clusters").as<uint16_t>();
```

Comment:
Require validation that the chosen values of `base_port`, `cluster_span`, `node_span`, `max_nodes_per_cluster`, and `max_clusters` are consistent in avoid port overlaps or exceeding the limit of `uint16_t`.